apps/molgenis-components/src/components/forms/BaseInput.vue

```diff
@@ -2,24 +2,44 @@
 export default {
   name: 'BaseInput',
   props: {
+    /**
+     * Unique identifier https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id
+     * Has no direct emx2 implementenation but can be constructend by concatinating the table name with the field name
+     */
     id: {
       type: String,
       required: true
     },
+    /**
+     * Name of the element. For example used by the server to identify the fields in form submits.
+     * relates to emx 2 columnName
+     */
     name: {
       type: String,
       required: false,
       default: function () { return this.id }
     },
     value: {type: [String, Number, Object, Array, Boolean], default: null},
+    /**
+     * Caption for an item in a user interface.
+     * relates to emx 2 columnName
+     */
     label: {
       type: String,
       required: false
     },
+    /**
+     * Additional information describing the field 
+     * relates to emx 2 description
+     */
     helpText: {
       type: String,
       required: false
     },
+    /**
+     * A string specifying the type of control to render. 
+     * (note: no direct relation to emx2 type, emx2 types are implemented as vue components)
+     */
     type: {
```

Comment:
so this is not related to columnType?

Reply:
correct, the base component type allows the extending types to set the html input type attr , i would suggest having vue components that have a ( non 1-to-1) mapping with emx fields ( like implemented in styleguide dist). at the html level there is still the need to use the html type attr to communicate with the browsers. ( example , the InputString uses the BaseInput.type to set the html input tag to type="text")

Reply:
ok, that makes sense.